From 51be9d12f3b9b48deb17c164bbe62331d8eace5c Mon Sep 17 00:00:00 2001
From: Bryan Gardiner <bog@khumba.net>
Date: Fri, 8 Dec 2023 05:59:56 -0800
Subject: [PATCH] Patch to find the data directory under /usr/share.

---
 Source/Boot.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/Boot.cpp b/Source/Boot.cpp
index 58e3990..f1b66d5 100644
--- a/Source/Boot.cpp
+++ b/Source/Boot.cpp
@@ -35,7 +35,7 @@ extern "C"
 
 static fs::path FindGameData()
 {
-	fs::path dataPath;
+    fs::path dataPath;
 
 #if __APPLE__
 	char pathbuf[PROC_PIDPATHINFO_MAXSIZE];
@@ -50,7 +50,7 @@ static fs::path FindGameData()
 	dataPath = pathbuf;
 	dataPath = dataPath.parent_path().parent_path() / "Resources";
 #else
-	dataPath = "Data";
+	dataPath = "/usr/share/Nanosaur2";
 #endif
 
 	dataPath = dataPath.lexically_normal();
-- 
2.41.0

