From 71dd42b0704137bbded71691a3b4072352ce1620 Mon Sep 17 00:00:00 2001
From: Bryan Gardiner <bog@khumba.net>
Date: Thu, 28 Mar 2024 19:03:26 -0700
Subject: [PATCH] Patch to find the data directory under /usr/share.

---
 CMakeLists.txt |  5 ++---
 src/Main.cpp   | 36 ++----------------------------------
 2 files changed, 4 insertions(+), 37 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c981edb..b7b0205 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,8 +61,6 @@ set(POMME_NO_MP3 true)
 # DEPENDENCIES
 #------------------------------------------------------------------------------
 
-add_subdirectory(extern/Pomme)
-
 find_package(SDL2 REQUIRED COMPONENTS main)
 find_package(OpenGL REQUIRED)  # only actually required if we want to force GLRENDER
 find_package(Threads REQUIRED)
@@ -124,7 +122,7 @@ endif()
 target_include_directories(${GAME_TARGET} PRIVATE
 	${SDL2_INCLUDE_DIRS}
 	${OPENGL_INCLUDE_DIR}
-	extern/Pomme/src
+	/usr/include/Pomme-@POMME_VER@
 	${GAME_SRCDIR}/Headers
 )
 
@@ -197,6 +195,7 @@ else()
 	set_target_properties(${GAME_TARGET} PROPERTIES COMPILE_DEFINITIONS_DEBUG "_CONSOLE")
 endif()
 
+target_link_directories(${GAME_TARGET} PRIVATE @POMME_LIBDIR@)
 target_link_libraries(${GAME_TARGET} ${GAME_LIBRARIES})
 
 #------------------------------------------------------------------------------
diff --git a/src/Main.cpp b/src/Main.cpp
index db57e76..bd6043b 100644
--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -27,39 +27,7 @@ extern "C"
 
 static fs::path FindGameData(const char* executablePath)
 {
-	fs::path dataPath;
-
-	int attemptNum = 0;
-
-#if !(__APPLE__)
-	attemptNum++;		// skip macOS special case #0
-#endif
-
-	if (!executablePath)
-		attemptNum = 2;
-
-tryAgain:
-	switch (attemptNum)
-	{
-		case 0:			// special case for macOS app bundles
-			dataPath = executablePath;
-			dataPath = dataPath.parent_path().parent_path() / "Resources";
-			break;
-
-		case 1:
-			dataPath = executablePath;
-			dataPath = dataPath.parent_path() / "Data";
-			break;
-
-		case 2:
-			dataPath = "Data";
-			break;
-
-		default:
-			throw std::runtime_error("Couldn't find the Data folder.");
-	}
-
-	attemptNum++;
+	fs::path dataPath("/usr/share/MightyMike");
 
 	dataPath = dataPath.lexically_normal();
 
@@ -72,7 +40,7 @@ tryAgain:
 
 	if (resFileRefNum == -1)
 	{
-		goto tryAgain;
+		throw std::runtime_error("Error loading resources from the Data folder.");
 	}
 
 	UseResFile(resFileRefNum);
-- 
2.43.2

