From 438893808d6de12aaf3827d88ec0ae7e1170037e Mon Sep 17 00:00:00 2001
From: Bryan Gardiner <bog@khumba.net>
Date: Fri, 11 Jun 2021 16:52:46 -0700
Subject: [PATCH] Patch to use system-wide directories.

---
 CMakeLists.txt | 14 ++------------
 src/Main.cpp   | 18 +-----------------
 2 files changed, 3 insertions(+), 29 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 249f0bb..b8cece3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,9 +34,6 @@ set(POMME_NO_QD3D true)  # This version of the game uses Quesa rather than Pomme
 # DEPENDENCIES
 #------------------------------------------------------------------------------
 
-add_subdirectory(extern/Quesa)
-add_subdirectory(extern/Pomme)
-
 find_package(SDL2 REQUIRED COMPONENTS main)
 find_package(OpenGL REQUIRED)
 
@@ -256,7 +253,8 @@ endif()
 target_include_directories(${GAME_TARGET} PRIVATE
 	${SDL2_INCLUDE_DIRS}
 	${OPENGL_INCLUDE_DIR}
-	extern/Pomme/src
+	/usr/include/Pomme
+	/usr/include/Quesa
 	${GAME_SRCDIR}
 	${GAME_SRCDIR}/Headers
 	${GAME_SRCDIR}/Headers/Glue
@@ -273,14 +271,6 @@ add_compile_definitions(
 	"$<$<CONFIG:DEBUG>:_DEBUG>"
 )
 
-target_compile_definitions(Quesa_SDL PUBLIC
-	QUESA_OS_SDL
-	QUESA_APPLY_FOG_COLOR_IN_ALL_PASSES
-	QUESA_NO_TRANSPARENCY_SORT
-	#"$<$<CONFIG:DEBUG>:Q3_DEBUG>"
-	"$<$<CONFIG:DEBUG>:QUESA_DUMP_SHADERS>"
-)
-
 target_compile_definitions(${GAME_TARGET} PRIVATE
 	QUESA_OS_SDL
 )
diff --git a/src/Main.cpp b/src/Main.cpp
index 4b3e88c..c002755 100644
--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -45,23 +45,7 @@ extern "C"
 
 static fs::path FindGameData()
 {
-	fs::path dataPath;
-
-#if __APPLE__
-	char pathbuf[PROC_PIDPATHINFO_MAXSIZE];
-
-	pid_t pid = getpid();
-	int ret = proc_pidpath(pid, pathbuf, sizeof(pathbuf));
-	if (ret <= 0)
-	{
-		throw std::runtime_error(std::string(__func__) + ": proc_pidpath failed: " + std::string(strerror(errno)));
-	}
-
-	dataPath = pathbuf;
-	dataPath = dataPath.parent_path().parent_path() / "Resources";
-#else
-	dataPath = "Data";
-#endif
+	fs::path dataPath("/usr/share/Bugdom");
 
 	dataPath = dataPath.lexically_normal();
 
-- 
2.31.1

