From 38605c306abc001f3e69d4d64822cd57797f036a Mon Sep 17 00:00:00 2001
From: Bryan Gardiner <bog@khumba.net>
Date: Mon, 6 Sep 2021 08:57:11 -0700
Subject: [PATCH] Fix old reference to Sound/cmixer.h.

---
 src/PommeVideo.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/PommeVideo.h b/src/PommeVideo.h
index 851081a..a120c77 100644
--- a/src/PommeVideo.h
+++ b/src/PommeVideo.h
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "PommeTypes.h"
-#include "Sound/cmixer.h"
+#include "SoundMixer/cmixer.h"
 
 #include <istream>
 #include <queue>
-- 
2.32.0

