From dc6de8c414ae21516b72daead79c8db88309b102 Mon Sep 17 00:00:00 2001
From: Bryan Gardiner <bog@khumba.net>
Date: Sat, 27 Apr 2019 13:25:55 -0700
Subject: [PATCH] Add missing QAction include

---
 src/prefspaneldialog.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/prefspaneldialog.cpp b/src/prefspaneldialog.cpp
index 6c0ee41..d280eb0 100644
--- a/src/prefspaneldialog.cpp
+++ b/src/prefspaneldialog.cpp
@@ -15,6 +15,7 @@
 #include "hyphenate/fmhyphenator.h"
 #include "fmpaths.h"
 
+#include <QAction>
 #include <QDebug>
 #include <QToolTip>
 #include <QSettings>
-- 
2.21.0

